packages/server-core/src/media/static-resource/static-resource.ts

```diff
@@ -57,4 +67,65 @@ export default (app: Application): void => {
 
   const service = app.service(staticResourcePath)
   service.hooks(hooks)
+
+  const onCRUD =
+    (app: Application) => async (data: StaticResourceType | Paginated<StaticResourceType> | StaticResourceType[]) => {
+      const projectNames: string[] = []
+      if (Array.isArray(data)) {
+        data.forEach((item) => {
+          if (item.project && item.type === 'scene') {
+            projectNames.push(item.project)
+          }
+        })
+      } else if ('data' in data) {
+        data.data.forEach((item) => {
+          if (item.project && item.type === 'scene') {
+            projectNames.push(item.project)
+          }
+        })
+      } else if (data.project && data.type === 'scene') {
+        projectNames.push(data.project)
+      }
+
+      const projects = await app.service(projectPath).find({
+        query: {
+          name: { $in: projectNames },
+          $select: ['id']
+        },
+        paginate: false
+      })
+
+      const targetIds: string[] = []
```

Comment:
is this going to be sending all static resources to all users who have permission? or sending only each of the static resources to those who have permission for each project?

Reply:
So, this is what it's doing:
- Fetch which projects this static-resource belongs to (can be multiple because, static-resource can be array)
- Fetch users who have access to these projects
- Publish event to all these users.

Reply:
right, so if the event has an array, and that array contains resources from multiple projects, are we sending all the resources to all users who have permission for any of the projects, or sending only each resource in the array to the users who have permission?

Reply:
Interesting! Right now we're sending all data to all captured users. But:
> sending only each resource in the array to the users who have permission

this makes more sense.